### PR TITLE
Port to typescript

### PR DESCRIPTION
**Pull request by TobidieTopfpflanze**

### CODE REVIEW

**Review by HerryYT — DISMISSED** [review]
Please, always use accessors for properties declared in the class (also for constructors). (e.g `private id: type = 0;`)

File: src/command/Command.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import CommandParameter from '../network/type/CommandParameter';
 import Player from '../player/Player';
-const CommandData = require('../network/type/command-data');
+import CommandData from '../network/type/CommandData';
```
Comment (Bavfalcon9) [inline src/command/Command.ts:3]: Import as type

File: src/command/Command.ts [hunk 2]
```diff
@@ -11,13 +11,15 @@ interface CommandProps {
...
 
-export default class Command extends CommandData {
-    id: string;
-    description?: string;
-    aliases?: Array<string>;
-    flags?: number;
-    permission?: string;
-    parameters?: Array<Set<CommandParameter>> | Set<CommandParameter>;
+export default class Command {
+    id: string = '';
```
Comment (Bavfalcon9) [inline src/command/Command.ts:15]: This is redundant as this is set in construct.

File: src/command/Command.ts [hunk 3]
```diff
@@ -11,13 +11,15 @@ interface CommandProps {
...
+    description?: string = '';
```
Comment (Bavfalcon9) [inline src/command/Command.ts:16]: Description is optional and should not have a preset value.

File: src/command/Command.ts [hunk 4]
```diff
@@ -11,13 +11,15 @@ interface CommandProps {
...
+    aliases?: Array<string> = [];
```
Comment (Bavfalcon9) [inline src/command/Command.ts:17]: Same here, this is optional and should not have a preset value.

File: src/network/packet/AvailableCommandsPacket.ts [hunk 5]
```diff
@@ -45,7 +45,7 @@ export default class AvailableCommandsPacket extends DataPacket {
 
             // Flags
             this.writeByte(data.flags);
-            this.writeByte(data.permission);
+            this.writeByte(data.permission as any);
```
Comment (Bavfalcon9) [inline src/network/packet/AvailableCommandsPacket.ts:48]: Should be numeric?

File: src/network/type/NetworkTransaction.ts [hunk 6]
```diff
@@ -0,0 +1,50 @@
+import Prismarine from '../../Prismarine';
```
Comment (Bavfalcon9) [inline src/network/type/NetworkTransaction.ts:1]: Import not used.

File: src/network/type/NetworkTransaction.ts [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+import PacketBinaryStream from '../PacketBinaryStream';
```
Comment (Bavfalcon9) [inline src/network/type/NetworkTransaction.ts:2]: Import as type

File: src/network/type/itemStackRequest/ConsumeStack.ts [hunk 8]
```diff
@@ -0,0 +1,19 @@
+import ItemStackRequestSlotInfo from './ItemStackRequestSlotInfo';
```
Comment (Bavfalcon9) [inline src/network/type/itemStackRequest/ConsumeStack.ts:1]: Import as type

File: src/network/type/itemStackRequest/Destroy.ts [hunk 9]
```diff
@@ -0,0 +1,19 @@
+import ItemStackRequestSlotInfo from './ItemStackRequestSlotInfo';
```
Comment (Bavfalcon9) [inline src/network/type/itemStackRequest/Destroy.ts:1]: Import as type

File: src/network/type/itemStackRequest/Drop.ts [hunk 10]
```diff
@@ -0,0 +1,23 @@
+import ItemStackRequestSlotInfo from './ItemStackRequestSlotInfo';
```
Comment (Bavfalcon9) [inline src/network/type/itemStackRequest/Drop.ts:1]: Import as type

File: src/network/type/itemStackRequest/Place.ts [hunk 11]
```diff
@@ -0,0 +1,23 @@
+import ItemStackRequestSlotInfo from './ItemStackRequestSlotInfo';
```
Comment (Bavfalcon9) [inline src/network/type/itemStackRequest/Place.ts:1]: Import as type

File: src/network/type/itemStackRequest/Swap.ts [hunk 12]
```diff
@@ -0,0 +1,19 @@
+import ItemStackRequestSlotInfo from './ItemStackRequestSlotInfo';
```
Comment (Bavfalcon9) [inline src/network/type/itemStackRequest/Swap.ts:1]: Import as type

File: src/network/type/itemStackRequest/Take.ts [hunk 13]
```diff
@@ -0,0 +1,23 @@
+import ItemStackRequestSlotInfo from './ItemStackRequestSlotInfo';
```
Comment (Bavfalcon9) [inline src/network/type/itemStackRequest/Take.ts:1]: Import as type

File: src/network/type/itemStackRequest/Take.ts [hunk 14]
```diff
@@ -0,0 +1,23 @@
...
+
+class Take {
+    public count: number;
+    public from: ItemStackRequestSlotInfo;
+    public to: ItemStackRequestSlotInfo;
+
+    constructor({
+        count,
```
Comment (Bavfalcon9) [inline src/network/type/itemStackRequest/Take.ts:9]: An interface may look nicer to implement here.

**Review by Bavfalcon9 — CHANGES_REQUESTED** [review]
I found a few things I think should be changed. If you disagree, please let me know :)

File: src/network/packet/AvailableCommandsPacket.ts (hunk 5 above)
Comment (filfat) [reply src/network/packet/AvailableCommandsPacket.ts:48]: This might be a problem on jsbinaryutils end.

**Pull request merged**